src/controllers/MainController.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { DragDropContext, Draggable } from 'react-beautiful-dnd';
+import TaskLine from '../components/TaskLine';
+import { getActive, addTask, reorder } from '../redux/TasksSlice';
+import { writeToLocalStorage } from '../repository/repository';
+
+import Main from '../components/Main';
+import { SHOW_ALL } from '../FilterConstatnts';
+
+export default function MainController() {
+    const tasks = useSelector(getActive);
+    const filter = useSelector((state) => state.filter.filter);
+
+    const dispatch = useDispatch();
+
+    const [tempText, setTempText] = React.useState('');
+
+    React.useEffect(() => writeToLocalStorage(tasks), [tasks]);
```

Comment:
Please, move the localStorage logic to redux middleware or create actions with help of https://redux-toolkit.js.org/api/createAsyncThunk
Your current approach will not help easily replace data storage in the future (for example replace locastorage with requests to Backend)

Reply:
Do I get it right, that redux middleware will write to the localStorage before Action is performed? And I need it to be done after? As for now I moved writeToLocalStorage(tasks) into each action logic. Can that be a solution?